src/shell.js

```diff
@@ -270,7 +270,22 @@ if (ENVIRONMENT_IS_NODE) {
     console.error('The "worker_threads" module is not supported in this node.js build - perhaps a newer version is needed?');
     throw e;
   }
-  global.Worker = nodeWorkerThreads.Worker;
+  /**
+   * @constructor
+   * @param {string|URL} url
+   */
+  let NodeWorker = nodeWorkerThreads.Worker;
+  // Node requires data and file protocol urls to be URLs.
+  class Worker extends NodeWorker {
```

Comment:
ES6 classes are not yet used within the library sources, so I think this also requires an update to `settings.TRANSPILE_TO_ES5`.

<details>
  <summary>Details</summary>

```diff
--- a/emcc.py
+++ b/emcc.py
@@ -2176,13 +2176,14 @@ def phase_linker_setup(options, state, newargs):
     # Emscripten requires certain ES6 constructs by default in library code
     # - https://caniuse.com/let              : EDGE:12 FF:44 CHROME:49 SAFARI:11
     # - https://caniuse.com/const            : EDGE:12 FF:36 CHROME:49 SAFARI:11
+    # - https://caniuse.com/class            : EDGE:13 FF:45 CHROME:49 SAFARI:9
     # - https://caniuse.com/arrow-functions: : EDGE:12 FF:22 CHROME:45 SAFARI:10
     # - https://caniuse.com/mdn-javascript_builtins_object_assign:
     #                                          EDGE:12 FF:34 CHROME:45 SAFARI:9
     # Taking the highest requirements gives is our minimum:
-    #                             Max Version: EDGE:12 FF:44 CHROME:49 SAFARI:11
-    settings.TRANSPILE_TO_ES5 = (settings.MIN_EDGE_VERSION < 12 or
-                                 settings.MIN_FIREFOX_VERSION < 44 or
+    #                             Max Version: EDGE:13 FF:45 CHROME:49 SAFARI:11
+    settings.TRANSPILE_TO_ES5 = (settings.MIN_EDGE_VERSION < 13 or
+                                 settings.MIN_FIREFOX_VERSION < 45 or
                                  settings.MIN_CHROME_VERSION < 49 or
                                  settings.MIN_SAFARI_VERSION < 110000 or
                                  settings.MIN_IE_VERSION != 0x7FFFFFFF)
```
</details>

Reply:
Browser versions old enough to not support classes also don't support threads at all, so I don't think this should ever need to be polyfilled. Furthermore, the polyfilling doesn't actually work because it causes node to error out with `TypeError: Class constructor Worker cannot be invoked without 'new'` 🤯